tests/sub_group_mask/sub_group_mask_subscript.cpp

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Extension Conformance Test
+//
+//  Provides tests to check sub_group_mask operator[]
+//
+*******************************************************************************/
+
+#include "sub_group_mask_common.h"
+
+#define TEST_NAME sub_group_mask_subscript
+
+namespace TEST_NAMESPACE {
+
+using namespace sycl_cts;
+#ifdef SYCL_EXT_ONEAPI_SUB_GROUP_MASK
+
+struct check_result_subscript {
+  bool operator()(const sycl::ext::oneapi::sub_group_mask &sub_group_mask,
+                  const sycl::sub_group &) {
+    for (size_t N = 0; N < sub_group_mask.size(); N++) {
+      if (sub_group_mask[sycl::id(N)] != (N % 2 == 0)) return false;
+    }
+    return true;
```

Comment:
Does it make sense to scatter the test code among 9 files? Just 4-5 of 60 lines in this file are specific to the test, the rest are duplicated.